### DISCUSSION TITLE
How to replicate ExplorerPatcher toggle for "Automatically hide the taskbar"?

### DISCUSSION THREAD

**Discussion by ghost** [valinet/ExplorerPatcher#2730] [Q&A]
I'm trying to create an AutoHotKey script that will hide the taskbar when I am using Adobe apps, and show the taskbar for everything else.

If I use this kind of AHK script, I can make this happen:
```
#Requires AutoHotkey v1.1.35
#Persistent
Process Priority,, B
SetTimer Check, 1000
Check:
WinActive("A")
WinGet procPath, ProcessPath
WinGetClass winClass
If !InStr(procPath, "explorer.exe") || winClass = "CabinetWClass" {
 ToolTip % procPath
 adobe    := InStr(procPath, "\Adobe\") > 0
 autoHide := DllCall("Shell32\SHAppBarMessage", "UInt", 4 ; ABM_GETSTATE
           , "Ptr", &APPBARDATA, "Int")
 If (adobe ^ autoHide)
  taskbarAutoHide(adobe)
}
Return

taskbarAutoHide(autoHide:= True) {
 RunWait % "powershell -command ""&{$p='HKCU:SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\StuckRectsLegacy'"
         . ";$v=(Get-ItemProperty -Path $p).Settings;$v[8]="
         . (autoHide ? 3 : 2)
         . ";&Set-ItemProperty -Path $p -Name Settings -Value $v;&Stop-Process -f -ProcessName explorer}"""
        ,, "Hide"
 SoundBeep 800
 Sleep 10000
}
```

The problem is that the code that restarts the Explorer process is really clunky, takes a long time to finish, and messes up a few system tray icons (they disappear).

I've noticed that if I change the setting in the EP Properties panel:
![rundll32_Pe3fMA67i5](https://github.com/valinet/ExplorerPatcher/assets/108174507/f2f22d50-5835-4253-8821-d9d98b6c8481)
the setting changes instantly without a restart of explorer...

I have searched through the repository with GrepWin and found some files that I think are involved:
[settings10.reg](https://github.com/valinet/ExplorerPatcher/blob/master/ep_gui/resources/settings10.reg)
[settings.reg](https://github.com/valinet/ExplorerPatcher/blob/master/ep_gui/resources/settings.reg)
[GUI.c](https://github.com/valinet/ExplorerPatcher/blob/master/ep_gui/GUI.c)
[dllmain.c](https://github.com/valinet/ExplorerPatcher/blob/master/ExplorerPatcher/dllmain.c)

But I am not a programmer at this level. Does anyone know if there is a command that could be run that would replicate EP's behaviour with this particular setting? A command that I can put in my AHK script?

**Comment by SubhutiBottleneck** [comment]
You say the code to restart Explorer is clunky and takes a long time?  You are talking about these two lines in a batch file?:

taskkill /f /im explorer.exe
start explorer.exe

But otherwise, I don't know enough about it either to make it change without restarting Explorer.

**Reply by ghost** [reply]
I don't know how Valinet did it, but when you toggle "Automatically hide the taskbar" in the EP Taskbar settings panel, it smoothly hides or unhides the taskbar.

I ended up using 7+ Taskbar Tweaker, setting a hotkey for the taskbar autohide, and then I put that hotkey into an AHK function that would trigger based on certain programs being in focus. Works well enough, just not quite as smoothly or quickly.

**Reply by pyrates999** [reply]
He probably used the same method of when you go into taskbar settings, and set the taskbar to be autohiding.

**Comment by pyrates999** [comment]
The setting for whether the taskbar autohides or not was always applied right away in windows.
